### ISSUE TITLE
Get-LatestAdobeFlashUpdate 1809 do not return x64

### ISSUE THREAD

**Issue by helgefylla** [aaronparker/LatestUpdate#48]
# Expected Behavior
Example for x86 is working. Also Version 1607 for x64 etc..

PS C:\> Get-LatestAdobeFlashUpdate | Where-Object { ($_.Architecture -eq 'x86') -and ($_.Note -like '*Windows 10*1809*')}

KB           : KB4503308
Note         : 2019-06 Security Update for Adobe Flash Player for Windows 10 Version 1809 for x86-based Systems (KB4503
               308)
URL          : http://download.windowsupdate.com/c/msdownload/update/software/secu/2019/06/windows10.0-kb4503308-x86_4f
               c8ff515a3c3dc94c2a5aa85af287d54fc555c6.msu
Version      : 1809
Architecture : x86

# Current Behavior
Returning nothing
PS C:\> Get-LatestAdobeFlashUpdate | Where-Object { ($_.Architecture -eq 'x64') -and ($_.Note -like '*Windows 10*1809*')}
PS C:\>

Note! **1809 x64** Not listed when running only: Get-LatestAdobeFlashUpdate

# Failure Information (for bugs)

Please help provide information about the failure if this is a bug by issuing the command using the `-Verbose` command.

```
No Output !
```

## Context

Please provide any relevant information about your setup. This is important in case the issue is not reproducible except for under certain conditions.

* **LatestUpdate Version**: Use `Get-Module -ListAvailable LatestUpdate` = 3.0.131
* **PowerShell Version**: Use `Get-PSVersion` = See below
* **Operating System**: Windows 10 1809

PSVersion                      5.1.17763.503
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17763.503
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

## Failure Logs

Please include any relevant log snippets or files here.

**Comment by f0nt4** [comment]
The problem here is that it's on the 2nd page of the search results which are not captured by the module.
The Microsoft Update Catalog page, only returns 25 results per page, and searching for the KB has 27 results.

**Comment by helgefylla** [comment]
Thanks Fonta for clarifying the result.

Is the Get-LatestCumulativeUpdate and LatestServicingStackUpdate handled the same way, so they might fail as well ?

Would it be a workaround to implement a Version parameter like for the others ?
Get-LatestServicingStackUpdate -Version 1809 
Get-LatestCumulativeUpdate -Version 1809

I usually do a search on www.catalog.update.microsoft.com like below to narrow down the results.

https://www.catalog.update.microsoft.com/Search.aspx?q=adobe+x64+1809

**Comment by aaronparker** [comment]
Each function outputs a `Version` and `Architecture` property, so the following command line would work:

```
Get-LatestAdobeFlashUpdate | Where-Object { ($_.Architecture -eq 'x64') -and ($_.Version -eq '1809') }
```

The challenge, as stated above is that the page returns more than 25 results, so I need to find out how to return more from the query. Adding a `Version` parameter to the function would be possible. I was trying to keep the functions as simple as possible.

**Comment by f0nt4** [comment]
@aaronparker I tried to find some url addition to change the amount of results per page, but couldn't.
The only method might be adding `-OS` and then with a switch statement make the searchstring more specific.
Maybe you have some other ideas though.

Also my PR is still waiting for you :)

**Comment by aaronparker** [comment]
I've had a look at the page as well and could find anything either. I'll consider whether to add `-Version` and `-Architecture` parameters and trade off some simplicity.

**Comment by aaronparker** [comment]
> Also my PR is still waiting for you :)

A bit of travel and work lately, I'll get to it soon.

**Comment by aaronparker** [comment]
Addressed in commit 617a8e516dc9b422e628ed7c32807ab5fa0c9d6e

**Comment by aaronparker** [comment]
Fixed in commit e3009ecc0dcc2f457f0c788681ceceb65bbcecef (v3.0.147). `Get-LatestAdobeFlashUpdate` now uses `-OperatingSystem` and `-Version` parameters to filter searches. To return 1809 updates use `Get-LatestAdobeFlashUpdate -OperatingSystem Windows10 -Version 1809`.
